pkgs/top-level/release-lib.nix

```diff
@@ -43,6 +46,11 @@ rec {
   pkgs_x86_64_cygwin = allPackages { system = "x86_64-cygwin"; };
 
 
+  assertTrue = bool:
+    if bool
+    then pkgs.runCommand "evaluated-to-true" {} "touch $out"
+    else pkgs.runCommand "evaluated-to-false" {} "false";
```

Comment:
Note that this is not what I would call recursive nix, as the module system does not evaluate any derivation, it only use the interpreter with `nix-instanciate` command, and check for the output of the command.